BUG-REPORT.md

```diff
@@ -0,0 +1,16 @@
+There seems to a [bug](https://issues.jenkins-ci.org/browse/JENKINS-38406) in the git-plugin where in, if a build
+parameter is part of branch spec, the jenkins job checks out a wrong branch.
+
+Detailed info on the bug in the [jira ticket](https://issues.jenkins-ci.org/browse/JENKINS-38406). I reproduced the bug
+from an out-of-the-box jenkins created from this repo. I have the docker image pushed at puneethn/jenkins-git-bug:latest.
+
+````
+docker run -i --rm -p 8080:8080 -p 50000:50000 -t puneethn/jenkins-git-bug:latest
+````
+
+````
+username: admin
+password: password
+````
+
+Refer to the build parameter `sha1` and branch built for jobs 8 and 9.
```

Comment:
That docker instance has security enabled and is prompting for a user name and password.  You can either modify it to not require security, or enable users to "sign up" so that I can create my own account.

I may also need access to the Dockerfile you used to create that instance, so that I can evaluate puling it into my repository.

I am unlikely to accept this pull request as is, since the repository already includes 10+ bug verification jobs, and the file name BUG-REPORT.md is not descriptive enough for my needs.